google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/RetryingConnectionImpl.java

```diff
@@ -95,7 +101,8 @@ protected void doStart() {
   public void reinitialize() {
     try (CloseableMonitor.Hold h = connectionMonitor.enter()) {
       if (completed) return;
-      currentConnection = connectionFactory.New(streamFactory, this, initialRequest);
+      lastInitialRequest = initialRequestProvider.getInitialRequest();
```

Comment:
proposal: Instead, make this method `public void reinitialize(StreamRequestT initialRequest)`. Then you prevent adding another upcall.

Reply:
Done.

However, I had to move the first `reinitialize` call from `RetryingConnectionImpl.doStart()` up to the stream impl `start()` methods because they store the initial request. I think this is more intuitive anyway, even though there is some duplication.

Also reordered some fields and methods in the stream impl classes, for consistency.

Reply:
> However, I had to move the first reinitialize call from RetryingConnectionImpl.doStart() up to the stream impl start() methods because they store the initial request.

Actually, just realized that this will cause all stream initialization during startup to be blocking. Moved the first reinitialize() call back into RetryingConnectionImpl and passed the first initial request to the constructor.